File: Posts_Website_Client/src/typescript/dtos/PostDto.ts
import { CommentDto } from "./CommentDto"
import { UserDto } from "./UserDto"

export interface PostDto {
    id: string
    title: string
    content: string,
    publishedAt: string,
    user: UserDto
}

export interface PostDetailsDto extends PostDto {
    comments: CommentDto[]
}

export interface PostFormDto {
    title: string
    content: string
}

File: src/Dtos/PostDto.cs
namespace Posts_Website.Dtos
{
    public record class PostDto
    (
        Guid Id,
        string Title,
        string Content,
        string PublishedAt,
        UserDto User
    );

    public record class PostDetailsDto
    (
        Guid Id,
        string Title,
        string Content,
        string PublishedAt,
        UserDto User,
        CommentDto[] Comments
    );

    public record class PostFormDto 
    (
        string Title,
        string Content
    );
}